api/src/main/java/org/openmrs/logging/OpenmrsConfigurationFactory.java

```diff
@@ -53,33 +58,39 @@ public class OpenmrsConfigurationFactory extends ConfigurationFactory {
 	private static final org.slf4j.Logger log = LoggerFactory.getLogger(OpenmrsConfigurationFactory.class);
 	
 	public static final String[] SUFFIXES = new String[] { ".xml", ".yml", ".yaml", ".json", "*" };
-	
+
 	@Override
 	public Configuration getConfiguration(LoggerContext loggerContext, String name, URI configLocation) {
 		if (!isActive()) {
 			return null;
 		}
-		
-		// try to load the configuration from the application directory
+
+		// try to load the configuration from the application data directory
 		if (configLocation == null) {
-			for (File applicationDirectory : new File[] {
-				OpenmrsUtil.getDirectoryInApplicationDataDirectory("configuration"),
-				OpenmrsUtil.getApplicationDataDirectoryAsFile()
- 			}) {
-				for (String suffix : getSupportedTypes()) {
-					if (suffix.equals("*")) {
-						continue;
-					}
-					
-					File configFile = new File(applicationDirectory, getDefaultPrefix() + suffix);
-					if (configFile.exists() && configFile.canRead()) {
-						return super.getConfiguration(loggerContext, name, configFile.toURI());
+			List<File> configurationFiles = getConfigurationFiles();
+			if (!configurationFiles.isEmpty()) {
+				if (configurationFiles.size() == 1) {
+					System.out.println("Adding log4j2 configuration file: " + configurationFiles.get(0).getPath());
+					return super.getConfiguration(loggerContext, name, configurationFiles.get(0).toURI());
+				}
+				else {
+					List<AbstractConfiguration> abstractConfigurations = new ArrayList<>();
+					for (File configFile : configurationFiles) {
+						Configuration configuration = super.getConfiguration(loggerContext, name, configFile.toURI());
+						if (configuration instanceof AbstractConfiguration) {
+							System.out.println("Adding log4j2 configuration file: " + configFile.getPath());
+							abstractConfigurations.add((AbstractConfiguration) configuration);
+						}
+						else {
+							System.out.println("Unable to add log4j2 configuration file: " + configFile.getPath());
```

Comment:
Since this is an error, it might make sense to write it to stderr, e.g.,

```suggestion
							System.err.println("Unable to add log4j2 configuration file: " + configFile.getPath());
```

Reply:
Done.